docs/dotnet-five.md

```diff
@@ -0,0 +1,138 @@
+
+## Language updates
+
+With .NET 5, the .NET programming languages are continuing to improve.
+
+### C# updates
+
+C# will move forward in lock-step with .NET releases. Developers writing .NET 5 apps will have access to the latest C# version and features. .NET 5 is paired with C# 9. C# 9 brings many new features to the language, here are a few highlights:
+
+- Records: Immutable reference types that behave like value types, and introduce the new `with` keyword into the language.
+- Relational pattern matching: Extends pattern matching capabilities to relational operators for comparative evaluations and expressions, including logical patterns - new keywords `and`, `or`, and `not`.
+- Top-level statements: As a means for accelerating adoption and learning of C#, the `Main` method can be omitted and application as simple as the following is valid:
+
+   ```csharp
+   System.Console.Write("Hello world!");
+   ```
+
+- Functional pointers: Language constructs that expose intermediate language (IL) the following opcodes `ldftn` and `calli`.
+
+For more information on the available C# 9 features, see [What's new in C# 9](csharp/whats-new/csharp-9.md).
+
+#### Source generators
+
+In addition to some of the highlighted new C# features, source generators are making their way into developer projects. Source generators allow code that runs during compilation to inspect your program and produce additional files that are compiled together with the rest of your code.
+
+For more information on source generators, see [Introducing C# source generators](https://devblogs.microsoft.com/dotnet/introducing-c-source-generators) and [C# source generator samples](https://devblogs.microsoft.com/dotnet/new-c-source-generator-samples).
+
+### F# updates
+
+F# is the .NET functional programming language, and with .NET 5, developers have access to F# 5. Here are several new features of F# 5, for more information, see [Announcing F# 5](https://devblogs.microsoft.com/dotnet/announcing-net-5-0-preview-8).
+
+#### Interpolated strings
+
+Similar to interpolated string in C#, and even JavaScript - F# supports basic string interpolation.
+
+```fsharp
+let name = "David"
+let age = 36
+let message = $"{name} is {age} years old."
+```
+
+In addition to basic string interpolation, there is typed interpolation. With typed interpolation, a given type must match the format specifier.
+
+```fsharp
+let name = "David"
+let age = 36
+let message = $"%s{name} is %d{age} years old."
+```
+
+### Visual Basic updates
+
+Going forward, there is no plan to evolve Visual Basic as a language. This supports language stability and maintains compatibility between the .NET Core and .NET Framework versions of Visual Basic. Future features of .NET that require language changes may not be supported in Visual Basic.
+
+With .NET 5, Visual Basic support is extended to:
+
+- Class library
+- Console
+- Windows Forms
+- Windows Presentation Foundation (WPF)
+- Worker Service
+- ASP.NET Core Web API
```

Comment:
There is no template for this in `dotnet new` - cut?

Reply:
Ah, good catch... I'll dig around a bit, but this list is missing several and has some that do not look accurate. I'll update it.